ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -1258,6 +1256,32 @@ defineEarlyCutoff' doDiagnostics cmp key file mbOld mode action = do
         --  * creating bogus "file does not exists" diagnostics
         | otherwise = useWithoutDependency (GetModificationTime_ False) fp
 
+-- Note [Housekeeping rule cache and dirty key out side of hls-graph]
+-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+-- Hls-graph contains its own internal running state for each key in the shakeDatabase.
+-- Rule result cache and dirty key are in ShakeExtras that is not visible to the hls-graph
```

Comment:
It's not clear to me which field in ShakeExtras represents rule result cache.
Could you please mention it here?
```suggestion
-- ShakeExtrac contains ??? field (rule result cache) and dirtyKeys (keys that became dirty in between build sessions) that is not visible to the hls-graph
```